### PR TITLE
docs(framework) Add how-to guide for using JSON output with CLI

### PR DESCRIPTION
**Pull request by chongshenng**

### CODE REVIEW

File: framework/docs/source/how-to-use-json-output.rst [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+``flwr run`` JSON output
+------------------------
+
+The |flwr_run|_ command runs a Flower app. By default, the command prints the status of
+the app build and run process as follows:
+
+.. code-block:: bash
+
+    $ flwr run
```
Comment (jafermarq) [inline framework/docs/source/how-to-use-json-output.rst:33]: I wonder if there is value in adding a comment saying that the below assumes you are in the directory of the app (same applies to other commands)

File: framework/docs/source/how-to-use-json-output.rst [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      "success": false,
+      "error-message": "Loading project configuration... \nSuccess\n There is no `[missing]` federation declared in the `pyproject.toml`.\n The following federations were found:\n\nfed-existing-1\nfed-existing-2\n\n"
+    }
+
+``flwr ls`` JSON output
+-----------------------
+
+The |flwr_ls|_ command lists all the runs in the current project. By default, the
+command list the details of one provided run ID or all runs in a Flower federation in a
+tabular format:
```
Comment (jafermarq) [inline framework/docs/source/how-to-use-json-output.rst:83]:
```suggestion
The |flwr_ls|_ command lists all the runs in the current project. By default, the
command lists the details all runs in a Flower federation in a tabular format:
```

Comment (jafermarq) [reply framework/docs/source/how-to-use-json-output.rst:83]: I'm not sure if mentioning the ID topic is needed (given the examples shown below)

File: framework/docs/source/how-to-use-json-output.rst [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          "elapsed": "00:00:09",
+          "created-at": "2024-12-16 12:18:39Z",
+          "running-at": "2024-12-16 12:18:39Z",
+          "finished-at": "N/A"
+        },
+      ]
+    }
+
+If the command fails, the JSON output will return two fields, ``success`` and
+``error-message``, as shown in the :ref:`failed command <json_error_output>` example.
```
Comment (jafermarq) [inline framework/docs/source/how-to-use-json-output.rst:138]: worth indicating that the ``error-message`` isn't the same as the one shown earlier (which is specific to `flwr run`)? Or maybe rephrasing this mini paragraph slightly is enough. This comment also applies to the `flwr stop` setting below.

File: framework/docs/source/how-to-use-json-output.rst (hunk 2 above)
Comment (chongshenng) [reply framework/docs/source/how-to-use-json-output.rst:83]: I've removed this mention to the last paragraph of this section. I think it's still worth mentioning what the JSON output will be for one entry.

File: framework/docs/source/how-to-use-json-output.rst (hunk 1 above)
Comment (chongshenng) [reply framework/docs/source/how-to-use-json-output.rst:33]: Added, thanks.

File: framework/docs/source/how-to-use-json-output.rst [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+The `Flower CLIs <ref-api-cli.html>`_ come with a built-in JSON output mode. This mode
+is useful when you want to consume the output of a Flower CLI programmatically. For
+example, you might want to use the output of the ``flwr`` CLI in a script or a
+continuous integration pipeline.
+
+.. note::
+
+    The JSON output mode is currently only available when using the Flower CLIs with a
+    `SuperLink <ref-api-cli.html#flower-superlink>`_.
```
Comment (jafermarq) [inline framework/docs/source/how-to-use-json-output.rst:12]:
```suggestion
    `SuperLink <ref-api-cli.html#flower-superlink>`_. Learn more about the `SuperLink`
    in the `Flower Architecture Overview <explanation-flower-architecture.html>`_ page.
```

**Review by jafermarq — APPROVED** [review]

File: framework/docs/source/how-to-use-json-output.rst [hunk 5]
```diff
@@ -0,0 +1,194 @@
+Use JSON outputs for Flower CLIs
+================================
```
Comment (danieljanes) [inline framework/docs/source/how-to-use-json-output.rst:2]:
```suggestion
Use `flwr` CLI JSON output
==========================
```

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**